infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationClientChannelAdapter.java

```diff
@@ -70,81 +72,89 @@ public void connectAsync() {
                 ManagedChannel channel = ManagedChannelBuilder.forAddress(nodeLocator.getHost(), nodeLocator.getPort())
                         .usePlaintext()
                         .build();
-                channelMap.put(node.getEndpoint(), channel);
-                blockingStubMap.put(node.getEndpoint(), LogReplicationChannelGrpc.newBlockingStub(channel));
-                asyncStubMap.put(node.getEndpoint(), LogReplicationChannelGrpc.newStub(channel));
-                onConnectionUp(node.getEndpoint());
+                channelMap.put(node.getRealNodeId().toString(), channel);
+                blockingStubMap.put(node.getRealNodeId().toString(), LogReplicationChannelGrpc.newBlockingStub(channel));
+                asyncStubMap.put(node.getRealNodeId().toString(), LogReplicationChannelGrpc.newStub(channel));
+                onConnectionUp(node.getRealNodeId().toString());
             } catch (Exception e) {
-                onConnectionDown(node.getEndpoint());
+                onConnectionDown(node.getRealNodeId().toString());
             }
         }));
     }
 
     @Override
-    public void connectAsync(String endpoint) {
+    public void connectAsync(String nodeId) {
+        String endpoint = getRemoteClusterDescriptor().getNodesDescriptors()
+                .stream()
+                .filter(nodeDescriptor -> nodeDescriptor.getRealNodeId().toString().equals(nodeId))
+                .map(NodeDescriptor::getEndpoint)
+                .collect(Collectors.toList())
+                .stream()
+                .findFirst()
+                .orElseThrow(RuntimeException::new);
```

Comment:
should we add a message indicating the error? or it could be hard to understand...

Reply:
Done.